pwiz_tools/Skyline/Model/Lib/NistLibSpec.cs

```diff
@@ -975,16 +985,29 @@ private bool CreateCache(ILoadMonitor loader, IProgressStatus status, int percen
                                 irt = GetRetentionTime(match.Groups[1].Value, false);
                         }
 
-                        if (line.StartsWith(@"_EOF_"))
+                        if (line.StartsWith(@"_EOF_", StringComparison.InvariantCultureIgnoreCase)) // Case insensitive
                             ThrowIOException(lineCount, Resources.NistLibraryBase_CreateCache_Unexpected_end_of_file);
-                        else if (line.StartsWith(@"Name:"))
+                        else if (line.StartsWith(NAME, StringComparison.InvariantCultureIgnoreCase)) // Case insensitive
                             break;
                     }
 
-                    // Watch for strange formula specification as seen in MONA LipidBlast, e.g. "[C11H22NO4]+" seen with adduct [M]+
-                    if (formula != null && formula.StartsWith(@"["))
+                    if (formula != null)
                     {
-                        formula = formula.Substring(1).Split(']')[0];
+                        // Watch for strange formula specification as seen in MoNA
+                        if (formula.StartsWith(@"["))
+                        {
+                            // In MoNA LipidBlast we've seen "[C11H22NO4]+" associated with adduct [M]+
+                            formula = formula.Substring(1).Split(']')[0];
+                        }
+                        // In MoNA Fiehn-HILIC we see "C7H9N2O+" associated with adduct [M]+
+                        else if (formula.Contains(@"+"))

```

Comment:
Don't really need the Contains here. string.Split()[0] works no matter what in this case.

Reply:
Easy enough to change to 

                        // In MoNA Fiehn-HILIC we see "C7H9N2O+" associated with adduct [M]+ or "C7H9N2O-" with [M]-
                        else
                        {
                            formula = formula.Split('+', '-')[0];
                        }